runtime/src/main/java/com/flipkart/flux/deploymentunit/ExecutableRegistryPopulator.java

```diff
@@ -123,10 +124,16 @@ public void run() {
                         }
                     }
 
-                    String taskIdentifier = new MethodId(method).toString() + _VERSION + version;
+                    MethodId methodId = new MethodId(method);
+                    String taskIdentifier = methodId.toString() + _VERSION + version;
+
+                    /* get concurrency config for this task */
+                    Map<String, Object> taskConfig = deploymentUnit.getTaskConfiguration(methodId.getPrefix());
```

Comment:
Instead of having a method in DeploymentUnit and calling it, I think we can retrieve once and keep the configuration object here itself. 

org.apache.commons.configuration.Configuration taskConfiguration = deploymentUnit.getConfiguration().subset("taskconfig");

taskConfiguration.getProperty(methodId.getPrefix() + ".executionConcurrency")